core/drivers/stm32_gpio.c

```diff
@@ -862,6 +863,71 @@ static TEE_Result apply_rif_config(struct stm32_gpio_bank *bank,
 	return res;
 }
 
+/* Forward reference to stm32_gpio_set_conf_sec() defined below */
+static void stm32_gpio_set_conf_sec(struct stm32_gpio_bank *bank);
+
+static TEE_Result stm32_gpio_fw_configure(struct firewall_query *firewall)
+{
+	struct stm32_gpio_bank *bank = firewall->ctrl->priv;
+	uint32_t gpios_mask = 0;
+	bool secure = true;
+	unsigned int gpio_pos = 0;
+
+	assert(bank->sec_support);
+
+	if (firewall->arg_count != 1)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	if (bank->rif_cfg) {
+#ifdef CFG_STM32_RIF
+		gpio_pos = (firewall->args[0] & RIF_PER_ID_MASK) >>
+			   RIF_PER_ID_SHIFT;
+#else
+		panic("RIF support expected");
+#endif
+		gpios_mask = BIT(gpio_pos);
+
+		/* We're about to change a specific GPIO config */
+		bank->rif_cfg->access_mask[0] |= gpios_mask;
+
+		/*
+		 * Update bank RIF config with firewall configuration data
+		 * and apply it.
+		 */
+		stm32_rif_parse_cfg(firewall->args[0], bank->rif_cfg,
+				    GPIO_MAX_CID_SUPPORTED, bank->ngpios);
+		return apply_rif_config(bank, gpios_mask);
+	}
+
+	/*
+	 * Non RIF GPIO banks use TZPROT() bit mask (bits 0 to 15)
+	 * and GPIO_STM32_NSEC bit flag in the DT bindings.
+	 */
+	gpios_mask = firewall->args[0] & GENMASK_32(15, 0);
```

Comment:
In the non-RIF case, we allow multiple pin reconfiguration. We forbid that for RIF-capable banks. I think the function should have the same use, no?

Reply:
Yes this is what I meant. I think this should be mentionned in a comment somewhere

Reply:
Do you think the inline comment above (lines 1025/1026) is not enough?